src/app/core/ui/ui/ui.component.ts

```diff
@@ -127,9 +129,13 @@ export class UiComponent {
       (s) => (this.siteSettings = s),
     );
 
-    this.setupService
-      .waitForConfigReady(true)
-      .then((ready) => this.configReady$.next(ready));
+    if (this.configService.hasConfig()) {
+      this.configReady$.next(true);
+    } else {
+      this.setupService
+        .waitForConfigReady(true)
```

Comment:
I actually thought waitForConfigReady(false) would do this also, but that doesn't seem to hide the menu on initial init. Did you understand what our param in waitForConfigReady currently does? (let's add that to the JsDoc there)

Reply:
I think what’s happening is this:
We’re using waitForConfigReady(false), which checks if either:
- Config is ready (i.e., configUpdates emits a non-undefined config), or
- Sync is completed (syncState === COMPLETED)
But configUpdates$ is based on: `configUpdates = this.entityUpdated.pipe(shareReplay(1))` 
If no config has been emitted yet, then configUpdates is undefined,
Maybe that's why, then in this case it's waiting till sync is COMPLETED